roborazzi/src/main/java/com/github/takahirom/roborazzi/Roborazzi.kt

```diff
@@ -460,30 +479,26 @@ fun SemanticsNodeInteraction.captureComposeNode(
       ),
       roborazziOptions = roborazziOptions
     ) {
-      canvases.add(it)
+      canvases.addIfChanged(it, roborazziOptions)
+    }
+  }
+  val handler = Handler(Looper.getMainLooper())
+  val composeApplyObserver = Snapshot.registerApplyObserver { anies, snapshot ->
+    handler.postAtFrontOfQueue {
+      try {
+        capture()
+      } catch (e: Exception) {
+        // It seems there is no screen, so wait
```

Comment:
This seems quite unspecific, could it fail for other reasons?

Reply:
Good point. I fixed the comments, and it no longer uses Exception for handling exceptions.

https://github.com/takahirom/roborazzi/pull/52/commits/6045cfcd2a4fdf8a4a4f7d8d48ad62d0d0255869